cardano-crypto-class/src/Cardano/Crypto/DSIGN/SchnorrSecp256k1.hs

```diff
@@ -0,0 +1,296 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+-- According to the documentation for unsafePerformIO:
+-- 
+-- > Make sure that the either you switch off let-floating 
+-- > (-fno-full-laziness), or that the call to unsafePerformIO cannot float 
+-- > outside a lambda.
+--
+-- If we do not switch off let-floating, our calls to unsafeDupablePerformIO for
+-- FFI functions become nondeterministic in their behaviour when run with
+-- parallelism enabled (such as -with-rtsopts=-N), possibly yielding wrong
+-- answers on a range of tasks, including serialization.
+{-# OPTIONS_GHC -fno-full-laziness #-}
+
+module Cardano.Crypto.DSIGN.SchnorrSecp256k1 (
+  SchnorrSecp256k1DSIGN,
+  VerKeyDSIGN,
+  SignKeyDSIGN,
+  SigDSIGN
+  ) where
+
+import qualified Data.ByteString as BS
+import Data.ByteString.Unsafe (unsafePackCStringLen)
+import Data.Primitive.Ptr (copyPtr)
+import Crypto.Random (getRandomBytes)
+import Cardano.Crypto.Seed (runMonadRandomWithSeed)
+import Data.ByteString.Internal (toForeignPtr, memcmp)
+import Foreign.ForeignPtr (
+  ForeignPtr, 
+  withForeignPtr, 
+  mallocForeignPtrBytes,
+  plusForeignPtr, 
+  castForeignPtr
+  )
+import Control.Monad (when)
+import System.IO.Unsafe (unsafeDupablePerformIO, unsafePerformIO)
+import Cardano.Binary (FromCBOR (fromCBOR), ToCBOR (toCBOR, encodedSizeExpr))
+import Foreign.Ptr (Ptr, castPtr, nullPtr)
+import Foreign.C.Types (CUChar)
+import Foreign.Marshal.Alloc (allocaBytes)
+import Cardano.Crypto.Schnorr (
+  SECP256k1XOnlyPubKey, 
+  secpKeyPairCreate, 
+  SECP256k1Context, 
+  secpKeyPairXOnlyPub,
+  SECP256k1SecKey, 
+  secpSchnorrSigVerify, 
+  secpContextSignVerify,
+  SECP256k1SchnorrSig, 
+  secpSchnorrSigSignCustom, 
+  secpContextCreate
```

Comment:
Some reason why some of these functions contain `Schnorr`, and other don't? or why some contain `secp` and others `SECP256k1`? And a nit, have we changed the order of the prefixes in purpose? In here we go `secp256k1schnorr` while above we go with `schnorrsecp256k1`.

Reply:
This is because they mirror the naming of the C functions they wrap (without `256k1` everywhere, admittedly).